warehouse/tuf/repository.py

```diff
@@ -0,0 +1,497 @@
+
+        meta = {"targets.json": MetaFile(targets.version)}
+        snapshot = Snapshot(1, SPEC_VERSION, payload[Snapshot.type].expiration, meta)
+        snapshot_metadata = Metadata(snapshot, {})
+        top_level_roles_metadata[Snapshot.type] = snapshot_metadata
+
+        snapshot_meta = MetaFile(snapshot.version)
+        timestamp = Timestamp(
+            1, SPEC_VERSION, payload[Timestamp.type].expiration, snapshot_meta
+        )
+        timestamp_metadata = Metadata(timestamp, {})
+        top_level_roles_metadata[Timestamp.type] = timestamp_metadata
+
+        roles = {
+            role_name: Role([], payload[role_name].threshold)
+            for role_name in TOP_LEVEL_ROLE_NAMES
+        }
+        root = Root(1, SPEC_VERSION, payload[Root.type].expiration, {}, roles, True)
+
+        # Sign all top level roles metadata
+        signers = dict()
+        for role in TOP_LEVEL_ROLE_NAMES:
+            if payload[role].threshold < len(payload[role].keys):
+                raise ValueError(
+                    f"Role {role} has missing Key(s) "
+                    f"to match to defined threshold {payload[role].threshold}."
+                )
+
+            for key in payload[role].keys:
+                root.add_key(role, Key.from_securesystemslib_key(key))
+
+            signers[role] = {
+                key["keyid"]: SSlibSigner(key) for key in payload[role].keys
+            }
+
+        root_metadata = Metadata(root, {})
+        top_level_roles_metadata[Root.type] = root_metadata
+        for role in signers:
+            for signer in signers[role].values():
+                top_level_roles_metadata[role].sign(signer, append=True)
+
+            if store:
+                self._store(role, top_level_roles_metadata[role])
+
+        self._is_initialized = True
+        return top_level_roles_metadata
+
+    def load_role(self, rolename: str) -> Metadata:
+        """
+        Loads a specific metadata by role name.
+
+        Args:
+            rolename: Role name
+
+        Returns:
+             Role Metadata
+             ``tuf.api.metadata.Metadata``
+        """
+        return Metadata.from_file(rolename, None, self.storage_backend)
+
+    def delegate_targets_roles(
+        self,
+        payload: Dict[str, List[RolesPayload]],
+    ) -> Metadata[Snapshot]:
+        """
+        Delegates a list of targets roles.
+        It will create the new role based in the parameters, sign and store,
+        updating the Delegator role, Snapshot and Timestamp roles.
+
+        Args:
+            payload: Dictionary of delegators with the delegated role parameters
+                     as RolePayload
+            timestamp_expires: New Timestamp expiration datetime
+            snapshot_expires: New Snapshot expiration datetime
+        """
+
+        snapshot_metadata = self.load_role(Snapshot.type)
+        for delegator, delegate_role_parameters in payload.items():
```

Comment:
May I ask, why payload is a list? Currently, you use this only with payloads of length 1.  Also see https://github.com/lukpueh/warehouse/pull/6#discussion_r818624051